### DISCUSSION TITLE
How to set inputs programmatically to have them properly saved?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#17294] [Archive]
In a custom interface `custominterface` input I set the initial value calculated from other retrieved values.
`input.value = 'Calculated value'`
if I don't modify by hand, it is not sent in the update and not saved at its calculated value.
How can I emit from `custominterface` to have it saved properly?

**Comment by jsx101** [comment]
Hi, this is my first time replying to a question but am happy to help.

Have you tried using `emit('input', insert_calculated_value_here)` in the script section once the calculation is complete? It should automatically change the value in the input field in the UI to your calculated value, and when saving, it will save with that value.

**Reply by Pachat** [reply]
Well you are right and thank you for answer. You are right: it works.
